### ISSUE TITLE
Dynamic disc not working on server.

### ISSUE THREAD

**Issue by Seikm** [Rubydesic/minecraft-dynamic-discs#7]
The mod works perfectly on singleplayer, but when I put it on the server it's not working. The only error that appears on the console is just one line: 
[Server thread/ERROR]: java.lang.IllegalStateException: Not a JSON Object: null

**Comment by HyperlyBola** [comment]
Having the same issue but there's no real log I can really show? I'll try grabbing client logs, but idk if that'll help because it worked fine on singleplayer. 

`02.06 23:57:45 [Server] Server thread/ERROR java.io.IOException: Cannot run program "/dynamic-discs/youtube-dl": error=13, Permission denied`

**Comment by Rubydesic** [comment]
@HyperlyBola Please make a separate issue with your full server log. If it contains confidential information (probably not) either redact it or privately message it to me on GitHub or discord. Thanks

**Comment by Seikm** [comment]
@Rubydesic is there anything i can do to fix my problem? I really thought this mod was awesome and wanted to be able to use it on my server...

**Comment by Rubydesic** [comment]
@Seikm Your error should no longer occur in the latest version, can you update to 2.0.7 and see if it works?

**Comment by Seikm** [comment]
@Rubydesic this time the error message changed:
`[Server thread/ERROR]: java.io.IOException: Cannot run program "/home/container/dynamic-discs/youtube-dl": error=13, Permission denied`
I'm using a host, maybe the problem is because of that. Is there something I should do?

**Comment by HyperlyBola** [comment]
> @Rubydesic this time the error message changed:
> `[Server thread/ERROR]: java.io.IOException: Cannot run program "/home/container/dynamic-discs/youtube-dl": error=13, Permission denied`
> I'm using a host, maybe the problem is because of that. Is there something I should do?

I actually managed to somehow figure it out in the server, I think I figured it out by accident. Delete youtube-dl, and just to be safe, the cache as well and anything but the config. I too am using a host (ExtraVM they're gr8) but I didn't have to contact them at all.

**Comment by Rubydesic** [comment]
@Seikm Please send your whole log

**Comment by Seikm** [comment]
@Rubydesic I tested doing what @HyperlyBola said, and the error message was weird. Couldn't find a result for the song name, but I tested it with several different songs, and restarted the server several times. Here are the logs from when I restarted the server until I tested it: 
[Could not ffind a result for the search.txt](https://github.com/Rubydesic/minecraft-dynamic-discs/files/6601888/Could.not.ffind.a.result.for.the.search.txt)
I installed everything again and the error is "Permission denied" again. Here are the logs from when the server started:
[Permission denied.txt](https://github.com/Rubydesic/minecraft-dynamic-discs/files/6601891/Permission.denied.txt)

**Comment by HyperlyBola** [comment]
Did you delete youtube-dl from the server or client? Also maybe try like a random well known song name just to test

**Comment by Rubydesic** [comment]
Strange. potentially youtube is blocked on the server? I will look into it more

**Comment by Seikm** [comment]
@Rubydesic Sorry I didn't respond for a while, I was asking the host if they were blocking something. The answer was that they didn't block anything. @HyperlyBola I tried to do it on both of them, just on one of them, anyway, and it didn't work.

**Comment by Seikm** [comment]
I spent some time trying to solve this annoying bug, but until today I haven't been able to. I've tried everything anyone can imagine, the only thing I think it might be is something on my host, but they said there's nothing blocking. I've given up for now, if you need me to test something, please tell me.

**Comment by Rubydesic** [comment]
I think this problem is fixed in the latest version, reopen if it's not.
